scipy/sparse/_coo.py

```diff
@@ -99,23 +99,26 @@ def __init__(self, arg1, shape=None, dtype=None, copy=False):
 
     @property
     def row(self):
-        return self.indices[0]
+        if self.ndim > 1:
+            return self.indices[0]
+        return np.zeros(self.nnz, dtype=self.indices[0].dtype)
+
 
     @row.setter
     def row(self, new_row):
+        if self.ndim < 2:
+            raise ValueError('cannot set row attribute of a 1-dimensional sparse array')
         new_row = np.asarray(new_row, dtype=self.indices[0].dtype)
         self.indices = (new_row,) + self.indices[1:]
```

Comment:
I think there's a subtlety here that we need to get right: in this new system, "col" is always `indices[-1]`, "row" is `indices[-2]`, and any theoretical other dimensions keep going right to left.

So instead of using `indices[0]` for the row in the 2d case, I'd prefer if we consistently used negative indices. Then the final assignment here would become:

```
self.indices = self.indices[:-2] + (new_row, self.indices[-1])
```

Reply:
Good -- I agree that this is the way to nd in the long term. Adding ndim to each lookup (to handle the negative index) shouldn't be costly. And this would provide uniformity for handling these shape-related features of the arrays.